lib/ddq-backend-mysql.js

```diff
@@ -0,0 +1,390 @@
+     * @param {Function} callback
+     */
+    function heartbeat(ddqBackendInstance, recordId, callback) {
+        ddqBackendInstance.connection.query(`
+            UPDATE ??
+            SET heartbeatDate = NOW()
+            WHERE hash = ?
+                AND isProcessing = true
+                AND owner = ?`,
+            [
+                ddqBackendInstance.config.table,
+                recordId,
+                ddqBackendInstance.owner
+            ], callback
+        );
+    }
+
+
+    /**
+     * Updates a specific record so that it can be requeued.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {string} recordId
+     * @param {Function} callback
+     */
+    function requeue(ddqBackendInstance, recordId, callback) {
+        ddqBackendInstance.connection.query(`
+            UPDATE ??
+            SET owner = null,
+                isProcessing = false,
+                requeued = false
+            WHERE hash = ?`,
+            [
+                ddqBackendInstance.config.table,
+                recordId
+            ], callback
+        );
+    }
+
+
+    /**
+     * Deletes a specific record from the database. If unsuccessful, this will
+     * call the command to requeue the record.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {string} recordId
+     * @param {Function} callback
+     */
+    function remove(ddqBackendInstance, recordId, callback) {
+        ddqBackendInstance.connection.query(`
+            DELETE
+            FROM ??
+            WHERE hash = ?
+                AND requeued = false
+                AND owner = ?`,
+            [
+                ddqBackendInstance.config.table,
+                recordId,
+                ddqBackendInstance.owner
+            ], (err) => {
+                if (err) {
+                    requeue(ddqBackendInstance, recordId, callback);
+                } else {
+                    callback();
+                }
+            }
+        );
+    }
+
+
+    /**
+     * Polls for a random record with the minimum delay defined by the
+     * config.pollingDelayMs.
+     *
+     * @param {Object} ddqBackendInstance
+     */
+    function poll(ddqBackendInstance) {
+        /**
+         * Queries the database to get a random record. Emits that random record
+         * along with methods to handle it.
+         */
+        function checkNow() {
+            var wrappedMessage;
+
+            ddqBackendInstance.poller = null;
+
+            ddqBackendInstance.connection.query(`
+                SELECT *
+                FROM ??
+                WHERE isProcessing = false
+                    AND RAND() < 4 / (
+                        SELECT COUNT(*)
+                        FROM ??
+                        WHERE isProcessing = false
+                        AND topics IN (?)
+                        OR topics IS NULL
+                    )
+                    AND topics IN (?)
+                    OR topics IS NULL
```

Comment:
Should not be using both of these.

According to the style guide, `OR` goes on the same line as `AND`.

Anyway, you should be either using a list of topics or you should require NULL.  A NULL topic should not mean "show this message in all topics".  Your list of topics may include `null` so you could pick these messages up.